ecc/Makefile

```diff
@@ -1,56 +1,74 @@
-.PHONY: all
+GOFLAGS:=
+GO:=go $(GOFLAGS)
 
+ENV_VARS:=LD_LIBRARY_PATH=$(LD_LIBRARY_PATH):./enclave/lib
 PEER_NAME?=dev-jdoe
 CC_NAME?=ecc
-LD_LIB_PATH=$(LD_LIBRARY_PATH):./enclave/lib
 
+VSCC_SRC = vscc/ecc_validation_plugin.go vscc/ecc_validation_logic.go
+VSCC_OUT = ecc-vscc.so
+
+DOCKERFLAGS:=
+DOCKER:=docker $(DOCKERFLAGS)
+DOCKER_CONTAINER=$(PEER_NAME)-$(CC_NAME)-0
+DOCKER_IMAGE?=$$(docker images | grep -- -$(CC_NAME)- | awk '{print $$1;}')
+
+.PHONY: all
 all: build vscc-plugin
 
+.PHONY: build
 build:
-	go get golang.org/x/sync/semaphore
-	go get github.com/pkg/errors
-	go get github.com/golang/protobuf/proto
-	LD_LIBRARY_PATH=$(LD_LIB_PATH) go build
+	$(GO) get golang.org/x/sync/semaphore
+	$(GO) get github.com/pkg/errors
+	$(GO) get github.com/golang/protobuf/proto
+	$(ENV_VARS) $(GO) $@
 
-vscc-plugin:
-	go build -o ./ecc-vscc.so -buildmode=plugin vscc/ecc_validation_plugin.go vscc/ecc_validation_logic.go
+vscc-plugin: $(VSCC_SRC)
+	$(ENV_VARS) $(GO) build -o $(VSCC_OUT) -buildmode=plugin $^
 
+.PHONY: test
 test:
-	LD_LIBRARY_PATH=$(LD_LIB_PATH) go test -v
+	$(ENV_VARS) $(GO) $@ -v
 
+.PHONY: stress
 stress:
-	LD_LIBRARY_PATH=$(LD_LIB_PATH) go test -v -run TestEnclaveChaincode_Invoke_Auction
+	$(ENV_VARS) $(GO) test -v -run TestEnclaveChaincode_Invoke_Auction
 
+.PHONY: debug
 debug: 
-	LD_LIBRARY_PATH=$(LD_LIB_PATH) go test -c
-	LD_LIBRARY_PATH=$(LD_LIB_PATH) sgx-gdb ecc.test -d $GOROOT
+	$(ENV_VARS) $(GO) test -c
+	$(ENV_VARS) sgx-gdb ecc.test -d $GOROOT
 
+.PHONY: clean
 clean:
-	go clean
+	$(GO) clean
+
+.PHONY: clean-all
+clean-all: clean docker-clean
```

Comment:
Agree, I also think a single `clean` from the top-level is enough; it can be handy to have separate extra cleans, such as docker-clean. 

I will remove the deploy step and reference directly.